spec/visual/samples/xpath

```diff
@@ -0,0 +1,319 @@
+(: Comment (not nested) (: Comment (nested) :) :)
+
+(: XPATH 2.0 :)
+(: Examples from https://www.w3.org/TR/xpath20/ :)
```

Comment:
What are the copyright licence requirements (if any) on these examples?

Reply:
The W3C's recommendations link to their rules for [document use](https://www.w3.org/Consortium/Legal/2015/doc-license), which state (emphasis mine):

> In addition, "Code Components" —Web IDL in sections clearly marked as Web IDL; and W3C-defined markup (HTML, CSS, etc.) and **computer programming language code clearly marked as code examples**— are licensed under the [W3C Software License](http://www.w3.org/Consortium/Legal/copyright-software).

The [W3C Software License](https://en.wikipedia.org/wiki/W3C_Software_Notice_and_License) states that a notice and a copy of the license must be included, so I've done that.

Reply:
One example was from a research paper, which is a copyrighted work that I should not have used. To avoid the inclusion of that in the commit history of this project, I've rebased my branch to get rid of the commits that added it. I'm afraid that rebase messed up the PR interface a little bit on GitHub (they generally [do not recommend rebasing open PRs](https://help.github.com/en/articles/why-are-my-commits-in-the-wrong-order)), so I'm sorry about that.

Reply:
No worries about the rebasing. This will be squashed before it's merged so the history won't be part of the repo :)